backend/src/services/PUBLICTutorialService.ts

```diff
@@ -0,0 +1,149 @@
+
+    /**
+     * Gets a PUBIC tutorial by Id
+     *
+     * @async
+     * @param {string} collectionName collection name
+     * @returns {Promise<T>} an array of the given documents
+     */
+    public static getPublicTutById(collectionName: string, tutId: ObjectId): Promise<PublicTutorial | null> {
+        return new Promise((res, rej) => {
+            const publicTutProjection: FindOneOptions = {
+                projection: {
+                    name: 1,
+                    html: 1,
+                    category: 1,
+                    authorName: 1,
+                    isAvailable: 1
+                }
+            };
+            PublicTutorialService.db.collection(collectionName).findOne<IProjectedTutorial>({ _id: tutId }, publicTutProjection,
+                (err, result) => {
+                    if (err) {
+                        rej(err);
+                    }
+                    res(PublicTutorialService.parsePublicProjectTut(result));
+                }
+            );
+        });
+    }
+
+
+    /**
+     * Gets all tutorial cards in a given category
+     *
+     * @param {string} collectionName the collection name
+     * @param {string} queryCategory the category to query
+     */
+    public static getTutCardsInCategory(collectionName: string, queryCategory: string): Promise<TutorialCard[]> {
+        return new Promise((res) => {
+            const tutCardProjection: FindOneOptions = {
+                projection: {
+                    _id: 1,
+                    name: 1,
+                    html: 1,
+                    category: 1,
+                    authorName: 1,
+                    isAvailable: 1
+                }
+            };
+            PublicTutorialService.db.collection(collectionName)
+                .find<IProjectedTutorial & { _id: string }>({ category: queryCategory }, tutCardProjection)
+                .toArray().then(
+                    (data) => {
+                        // Now we need to strip the HTML out of the html tag, and create a 'tutorial card'
+                        const responseCards: TutorialCard[] = data.map(preparsedCard => {
+                            const { _id, name, html, category, authorName, isAvailable } = preparsedCard;
+                            // Remove any stuff prior to first section, as it shouldn't be there
+                            let cardText = html.substring(html.indexOf("--##") + 4, html.length);
+                            // Remove all html related
+                            cardText = cardText.replace(/<\/?[^>]+(>|$)/g, "");
+                            // Remove section markers and give it a trim
+                            cardText = cardText.replace(/--##/g, "").trim();
+                            // Grab only 105 chars, may be subject to change
+                            cardText = cardText.substring(0, 105);
+                            return new TutorialCard(_id, name, category, cardText, authorName, isAvailable);
```

Comment:
I wonder If any can be done here. Some thing more declarative which will not require these many comments.
```js
cardText = cardText.replace(/<\/?[^>]+(>|$)/g, "");
                            // Remove section markers and give it a trim
                            cardText = cardText.replace(/--##/g, "").trim();
                            // Grab only 105 chars, may be subject to change
                            cardText = cardText.substring(0, 105);
```
these can be then merged in a single line ... yo ucan remove `--##` as well in your first regex using the pipe symbol.

Reply:
Nothing stopping us chaining replaces, no. But I wanted it to be explicitly clear what is happening, so for this reason, I'd advise against changing it and removing the comments.

Reply:
it's your choice. I'm not touching this file anyway :P .

Reply:
Yeah leave this one, so others understand exactly what's happening lol. The html removal regex isn't one we can read easily lol